packages/emails/src/index.tsx

```diff
@@ -42,15 +42,29 @@ export const DeliverEmailSchema = z
 
 export type DeliverEmail = z.infer<typeof DeliverEmailSchema>;
 
+export type NodeMailerTransportOptions = {
+  host?: string;
+  port?: number;
+  secure?: boolean;
+  auth?: {
+    user?: string;
+    pass?: string;
+  };
+};
+
 export class EmailClient {
-  #client?: Resend;
+  #client?: nodemailer.Transporter;
   #imagesBaseUrl: string;
   #from: string;
   #replyTo: string;
 
-  constructor(config: { apikey?: string; imagesBaseUrl: string; from: string; replyTo: string }) {
-    this.#client =
-      config.apikey && config.apikey.startsWith("re_") ? new Resend(config.apikey) : undefined;
+  constructor(config: {
+    smtpConfig?: NodeMailerTransportOptions;
+    imagesBaseUrl: string;
+    from: string;
+    replyTo: string;
+  }) {
+    this.#client = config.smtpConfig ? nodemailer.createTransport(config.smtpConfig) : undefined;
```

Comment:
this `smtpConfig` option is always going to be "truthy"  so the `nodemailer.createTransport` thing is always going to be called:

![CleanShot 2023-08-21 at 16 14 58](https://github.com/triggerdotdev/trigger.dev/assets/534/77f35e72-7760-4cd8-b611-a02e97c8aed1)

Reply:
I'm sorry but I don't quite understand this

Reply:
@ericallam  Should I put back the resend api key and dependency as a fallback to when the smtp config doesn't exist?

Reply:
I think Eric just meant the config values should be checked instead of the entire object. Otherwise, this will never return `undefined`. Checking for truthy `smtpConfig.host` should probably be enough.